optashift-employee-rostering-benchmark/src/main/resources/META-INF/persistence.xml

```diff
@@ -2,16 +2,26 @@
 <!-- OVERWRITES the normal persistence.xml in the server module -->
 <persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              version="2.1" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
-  <persistence-unit name="optashift-employee-rostering-persistence-unit">
+  <persistence-unit name="optashift-employee-rostering-persistence-unit" transaction-type="RESOURCE_LOCAL">
```

Comment:
This resource local might be the reason you're getting empty lists.
IIRC, resource local means you're say you will explicitly open and close all transactions everywhere in your code.
So the generated testdata might not get commited to the database.

Reply:
Changing it to JTA causes a NPE when building the entityManager (no other info is given).

Reply:
@Christopher-Chianelli for JTA transactions, I would suggest configuring a Datasource on the container - I can't see any used in your persistence.xml